src/app/context/CurrencyContext.tsx

```diff
@@ -0,0 +1,310 @@
+import { useState, useEffect, createContext, useContext } from "react";
+// https://www.npmjs.com/package/currency.js 0 dependencies, 135,987k weekly downloads
+import currencyJs from "currency.js";
+// https://github.com/MikeMcl/decimal.js-light 0 dependencies, 826,917 weekly downloads
+import { Decimal } from "decimal.js-light";
+import axios from "axios";
+
+import { useAuth } from "./AuthContext";
+import api from "~/common/lib/api";
+import { SupportedCurrencies, SupportedExchanges } from "~/types";
+
+interface CurrencyContextType {
+  balances: {
+    satsBalance: string;
+    fiatBalance: string;
+  };
+  setCurrencyValue: (currency: SupportedCurrencies) => void;
+  currencies: string[];
+  getFiatValue: (amount: number | string) => Promise<string>;
+}
+
+const CurrencyContext = createContext({} as CurrencyContextType);
+
+const numSatsInBtc = 100_000_000;
+
+export function CurrencyProvider({ children }: { children: React.ReactNode }) {
+  const currencies = [
+    "AED",
+    "AFN",
+    "ALL",
+    "AMD",
+    "ANG",
+    "AOA",
+    "ARS",
+    "AUD",
+    "AWG",
+    "AZN",
+    "BAM",
+    "BBD",
+    "BDT",
+    "BGN",
+    "BHD",
+    "BIF",
+    "BMD",
+    "BND",
+    "BOB",
+    "BRL",
+    "BSD",
+    "BTC",
+    "BTN",
+    "BWP",
+    "BYR",
+    "BZD",
+    "CAD",
+    "CDF",
+    "CHF",
+    "CLF",
+    "CLP",
+    "CNY",
+    "COP",
+    "CRC",
+    "CUP",
+    "CVE",
+    "CZK",
+    "DJF",
+    "DKK",
+    "DOP",
+    "DZD",
+    "EEK",
+    "EGP",
+    "ERN",
+    "ETB",
+    "EUR",
+    "FJD",
+    "FKP",
+    "GBP",
+    "GEL",
+    "GHS",
+    "GIP",
+    "GMD",
+    "GNF",
+    "GTQ",
+    "GYD",
+    "HKD",
+    "HNL",
+    "HRK",
+    "HTG",
+    "HUF",
+    "IDR",
+    "ILS",
+    "INR",
+    "IQD",
+    "IRR",
+    "ISK",
+    "JEP",
+    "JMD",
+    "JOD",
+    "JPY",
+    "KES",
+    "KGS",
+    "KHR",
+    "KMF",
+    "KPW",
+    "KRW",
+    "KWD",
+    "KYD",
+    "KZT",
+    "LAK",
+    "LBP",
+    "LKR",
+    "LRD",
+    "LSL",
+    "LTL",
+    "LVL",
+    "LYD",
+    "MAD",
+    "MDL",
+    "MGA",
+    "MKD",
+    "MMK",
+    "MNT",
+    "MOP",
+    "MRO",
+    "MTL",
+    "MUR",
+    "MVR",
+    "MWK",
+    "MXN",
+    "MYR",
+    "MZN",
+    "NAD",
+    "NGN",
+    "NIO",
+    "NOK",
+    "NPR",
+    "NZD",
+    "OMR",
+    "PAB",
+    "PEN",
+    "PGK",
+    "PHP",
+    "PKR",
+    "PLN",
+    "PYG",
+    "QAR",
+    "RON",
+    "RSD",
+    "RUB",
+    "RWF",
+    "SAR",
+    "SBD",
+    "SCR",
+    "SDG",
+    "SEK",
+    "SGD",
+    "SHP",
+    "SLL",
+    "SOS",
+    "SRD",
+    "STD",
+    "SVC",
+    "SYP",
+    "SZL",
+    "THB",
+    "TJS",
+    "TMT",
+    "TND",
+    "TOP",
+    "TRY",
+    "TTD",
+    "TWD",
+    "TZS",
+    "UAH",
+    "UGX",
+    "USD",
+    "UYU",
+    "UZS",
+    "VEF",
+    "VND",
+    "VUV",
+    "WST",
+    "XAF",
+    "XAG",
+    "XAU",
+    "XBT",
+    "XCD",
+    "XDR",
+    "XOF",
+    "XPF",
+    "YER",
+    "ZAR",
+    "ZMK",
+    "ZMW",
+    "ZWL",
+  ];
+
```

Comment:
I assume most of this is copied from the neglected lib. Maybe link it here so people understand where this is coming from? Even though we do not use it :P

Can we use the tests as well?  
https://github.com/AryanJ-NYC/bitcoin-conversion/blob/master/test/bitcoin-conversion.spec.ts

Reply:
Closed? Should we add tests later? I could support with that.